strawberry/enum.py

```diff
@@ -48,7 +64,19 @@ def _process_enum(
 
     description = description
 
-    values = [EnumValue(item.name, item.value) for item in cls]  # type: ignore
+    values = []
+    for item in cls:  # type: ignore
+        item_value = item.value
+        item_name = item.name
+        deprecation_reason = None
+
+        if isinstance(item_value, EnumValueDefinition):
+            deprecation_reason = item_value.deprecation_reason
+            item_value = item_value.value
+
+        values.append(
+            EnumValue(item_name, item_value, deprecation_reason=deprecation_reason)
+        )
```

Comment:
minor nit for slightly better readability + shorter:
```suggestion
value = EnumValue(item_name, item_value, deprecation_reason=deprecation_reason)
values.append(value)
```